ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala

```diff
@@ -59,56 +59,85 @@ object LedgerApiBenchTool {
     val names = new Names
     val authorizationHelper = config.authorizationTokenSecret.map(new AuthorizationHelper(_))
 
-    apiServicesOwner(config, authorizationHelper).use {
-      servicesForUserId: (String => LedgerApiServices) =>
-        def benchmarkStep(
-            streamConfigs: List[WorkflowConfig.StreamConfig]
-        ): Future[Either[String, Unit]] =
-          if (streamConfigs.isEmpty) {
-            logger.info(s"No streams defined. Skipping the benchmark step.")
-            Future.successful(Right(()))
-          } else {
-            Benchmark.run(
-              streamConfigs = streamConfigs,
-              reportingPeriod = config.reportingPeriod,
-              apiServices = servicesForUserId(names.benchtoolUserId),
-              metricsReporter = config.metricsReporter,
+    def regularUserCreationStep(adminServices: LedgerApiServices): Future[Unit] =
+      config.authorizationTokenSecret match {
+        case None => Future.successful(())
+        case Some(_) =>
+          // Only if the submission config is provided we can compute what parties the user can observe.
+          // Otherwise we set an empty list as a default not to prematurely block the stream reading phase.
```

Comment:
How the stream reading phase could be blocked?

Reply:
Discussed offline that the comment is not precise. Fixed in 60fc2a0bad45bbd845f9daf6f03f1b094f779c54